js/libs/builder.js

```diff
@@ -323,6 +323,44 @@ Fliplet().then(function() {
           Fliplet.Studio.emit('widget-mode', 'wide');
         }
       },
+      getDataSourceColumnValues: function(field) {
+        var id = field.dataSourceId;
+        var column = field.column;
+
+        if (!id || !column) {
+          return;
+        }
+
+        Fliplet.Cache.get({
+          key: id + '-' + column,
+          expire: 60
+        }, function getColumnValues() {
+          // If there's no cache, return new values, i.e.
+          Fliplet.DataSources.connect(id).then(function(connection) {
```

Comment:
This looks like a problem from the beginning, before your change, because the Promise request isn't being returned.

```suggestion
          return Fliplet.DataSources.connect(id).then(function(connection) {
```